<launch>

  <arg name="rviz"     default="true" doc="whether to show Rviz" />
  <param name="file_path" value="$(find stretch_core)/config" />
  
  <!-- STRETCH DRIVER -->
  <param name="/stretch_driver/broadcast_odom_tf" type="bool" value="true"/>
  <param name="/stretch_driver/mode" type="string" value="position" />
  <include file="$(find stretch_core)/launch/stretch_driver.launch" pass_all_args="true"/>

  <!-- CENTERED BASE LINK -->
  <node name="centered_base_link_tf_publisher" pkg="tf" type="static_transform_publisher" args="-0.1 0 0 0 0 0 1 /base_link /centered_base_link 100" />

  <!-- LASER RANGE FINDER -->
  <include file="$(find stretch_core)/launch/rplidar.launch" />

  <!-- ARUCO MARKER DETECTOR -->
  <rosparam command="load" file="$(find stretch_core)/config/stretch_marker_dict.yaml" />
  <node name="detect_aruco_markers" pkg="stretch_core" type="detect_aruco_markers" output="screen"/>

  <!-- MAPPING -->
  <node pkg="gmapping" type="slam_gmapping" name="gmapping" output="log" />

  <!-- NAVIGATION -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find stretch_navigation)/config/common_costmap_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find stretch_navigation)/config/common_costmap_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find stretch_navigation)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find stretch_navigation)/config/global_costmap_params_withmap.yaml" command="load" />
    <rosparam file="$(find stretch_navigation)/config/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find stretch_navigation)/config/base_global_planner_params.yaml" command="load" />
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <remap from="/cmd_vel" to="/stretch/cmd_vel" />
  </node>

  <!-- VISUALIZE -->
  <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find stretch_core)/rviz/stretch_aruco_test.rviz" if="$(arg rviz)" />

</launch>
